### DISCUSSION TITLE
btc_type Adding process

### DISCUSSION THREAD

**Discussion by Raz0rLegend** [Vdauphin/HeartsAndMinds#1348] [Question and answer]
Hi everyone ! 
Inside the mission.sqf to remove and add units, could i sum up some units ? 
for exemple, we will play the VDV Rhs facvtions. If i don't want the all flora, mflora and recon units, do i have to write all the classnames of all the units, or could i put ?

``` C++
SOF_Units = [];
private _CfgVehicles = configFile >> "CfgVehicles";
{
    private _unitFilter = format ["(configName _x find '%1' == 0) && (getNumber (_x >> 'scope') == 2)", _x];
    private _unitList = _unitFilter configClasses _CfgVehicles apply { configName _x };
    SOF_Units append _unitList;
} forEach [ 
    "rhs_vdv_flora",
    "rhs_vdv_mflora",
    "rhs_vdv_recon",
    "rhs_vdv_des"
];
btc_type_units = btc_type_units - ["_x"];
```

**Comment by Vdauphin** [comment]
Hello,

> do i have to write all the classnames of all the units,

Looks easier as the code you wrote at the end don't do that much

Cheers

**Reply by Raz0rLegend** [reply]
That's because there are so many soldiers to delete. :''(

**Reply by Vdauphin** [reply]
```sqf
btc_type_units = btc_type_units - SOF_Units;
```

Don't know if `SOF_Units` is fill correctly 🤷‍♂️

> That's because there are so many soldiers to delete. :''(

Years ago each faction needed to be type on by one.... https://github.com/Giallustio/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/mission.sqf#L410

**Reply by Raz0rLegend** [reply]
years after years we tried to do less. :D

**Reply by Raz0rLegend** [reply]
Hmmm In this case 
`
btc_type_units = btc_type_units - SOF_Units;`
I need to replace my _x by SOF_Units right ? 
 ```sqf
SOF_Units = [];
        private _CfgVehicles = configFile >> "CfgVehicles";
        {
            private _unitFilter = format ["(configName _x find '%1' == 0) && (getNumber (_x >> 'scope') == 2)", SOF_Units];
            private _unitList = _unitFilter configClasses _CfgVehicles apply { configName SOF_Units };
            SOF_Units append _unitList;
        } forEach [ 
            "rhs_vdv_flora",
            "rhs_vdv_mflora",
            "rhs_vdv_recon",
            "rhs_vdv_des"
        ];
```

**Reply by Vdauphin** [reply]
No `_x` is only define betwween the `{}` of `forEach` this is why you created the variable `SOF_Units`... Right?

Try the official Arma Discord server for help on sqf
